cmd/fanoutsidecar/main.go

```diff
@@ -162,6 +174,22 @@ func setupConfigMapWatcher(logger *zap.Logger, mgr manager.Manager, configUpdate
 		return err
 	}
 
-	mgr.Add(cmw)
+	err = mgr.Add(utils.NewBlockingStart(logger, cmw))
+	if err != nil {
+		logger.Error("Unable to add the config map watcher", zap.Error(err))
```

Comment:
If the error will be logged by the manager, the other option would be to simply write:

```golang
return mgr.Add(....)
```

Reply:
I'm not sure if the error will be logged by the manager. For code that is run once at startup, I am OK with redundant logging on a failure, I find it makes it easier to debug when I can clearly see the point in our code that saw the error first.